newIDE/app/src/MainFrame/EditorContainers/StartPage/Carousel.js

```diff
@@ -0,0 +1,338 @@
+        </>
+      );
+    },
+    [cellWidth, titleHeight]
+  );
+
+  const computeScroll = React.useCallback(
+    (
+      direction: 'left' | 'right',
+      scrollViewElement: HTMLUListElement
+    ): number => {
+      const visibleThumbnailsCount = Math.floor(
+        scrollViewElement.offsetWidth / widthUnit
+      );
+      const scale = visibleThumbnailsCount * widthUnit;
+
+      const currentScroll = scrollViewElement.scrollLeft;
+      const currentFirstVisibleItemIndex = currentScroll / widthUnit;
+
+      if (
+        currentFirstVisibleItemIndex >
+        itemsToDisplay.length - visibleThumbnailsCount - 1
+      )
+        return 0;
+      return roundScroll(
+        scrollViewElement.scrollLeft + scale * (direction === 'left' ? -1 : 1)
+      );
+    },
+    [widthUnit, itemsToDisplay]
+  );
+
+  const roundScroll = React.useCallback(
+    (value: number): number => {
+      return Math.round(value / widthUnit) * widthUnit;
+    },
+    [widthUnit]
+  );
+
+  const onClickArrow = (direction: 'left' | 'right') =>
+    React.useCallback(
+      (): void => {
+        const scrollViewElement = scrollView.current;
+        if (!scrollViewElement) return;
+        const newScrollPosition = computeScroll(direction, scrollViewElement);
+
+        scrollViewElement.scrollTo({
+          left: newScrollPosition,
+          behavior: 'smooth',
+        });
+      },
+      [computeScroll]
+    );
+
+  const handleScroll = React.useCallback(() => {
+    const scrollViewElement = scrollView.current;
+    if (!scrollViewElement) return;
+
+    if (!shouldDisplayLeftArrow)
+      setShouldDisplayLeftArrow(scrollViewElement.scrollLeft !== 0);
+  });
+  const handleScrollEnd = React.useCallback(
+    () => {
+      const scrollViewElement = scrollView.current;
+      if (!scrollViewElement) return;
+
+      scrollViewElement.scrollTo({
+        left: roundScroll(scrollViewElement.scrollLeft),
+        behavior: 'smooth',
+      });
+    },
+    [roundScroll]
+  );
+
+  React.useEffect(() => {
+    const scrollViewElement = scrollView.current;
+    if (!scrollViewElement) return;
+
+    scrollViewElement.addEventListener('scroll', handleScroll);
+    scrollViewElement.addEventListener('touchend', handleScrollEnd);
+    scrollViewElement.addEventListener('touchleave', handleScrollEnd);
+  }, []);
+
+  return (
+    <Line noMargin expand>
+      <div
+        style={{ ...styles.leftArrowContainer, width: arrowWidth }}
+        onClick={onClickArrow('left')}
+      >
+        {!!items && shouldDisplayLeftArrow && <ArrowBackIos />}
+      </div>
+      <div style={{ width: `calc(100% - ${arrowWidth}px)` }}>
+        <Text size="title">{title}</Text>
+        <GridList
+          cols={itemsToDisplay.length}
+          cellHeight={cellHeight}
+          spacing={cellSpacing}
+          style={styles.gridList}
+          ref={scrollView}
+        >
+          {itemsToDisplay.map(item => (
+            <GridListTile key={computeItemKey(item)}>
```

Comment:
Would be interesting to check if we can focus the tiles using the keyboard, for accessibility i think it would be great to support "tabbing" between them.

Reply:
I'll check!

Reply:
I tried to do something, specifying a tab index offset and adding tab index to tiles, it works but it's not perfect. We can perfect it later if the specs are clear.